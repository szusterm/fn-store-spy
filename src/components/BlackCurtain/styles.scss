@import '../../scss/transition';
@import '../../scss/opacity-colors';

.black-curtain {
	cursor: pointer;
	position: fixed;
	top: 0;
	bottom: 0;
	right: 0;
	left: 100%;

	z-index: 15;

	@include transition(md);
	transition-property: background-color;

	&--showed {
		left: 0;
		@include black-background(0.8);
	}
}