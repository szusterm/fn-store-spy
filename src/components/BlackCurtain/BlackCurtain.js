import React, {Component} from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

export class BlackCurtain extends Component {

}

export default BlackCurtain;