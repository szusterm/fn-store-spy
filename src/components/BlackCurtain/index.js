import BlackCurtain from './BlackCurtain';

export default BlackCurtain;