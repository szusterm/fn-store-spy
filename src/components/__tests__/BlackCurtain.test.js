import React from 'react';
import {shallow} from 'enzyme';

import {BlackCurtain} from '../BlackCurtain/BlackCurtain';

const setup = (propOverrides) => {
	const props = Object.assign({
		show: true,
		onClick: () => true
	}, propOverrides);

	const wrapper = shallow(<BlackCurtain {...props}/>);

	return {props, wrapper};
};

describe('CircleOrderCount Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});
});