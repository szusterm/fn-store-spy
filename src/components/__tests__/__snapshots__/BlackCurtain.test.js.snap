// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleOrderCount Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlackCurtain
    onClick={[Function]}
    show={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "black-curtain black-curtain--showed",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "black-curtain black-curtain--showed",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
